lnwallet/channel.go

```diff
@@ -3508,8 +3672,11 @@ func (lc *LightningChannel) ProcessChanSyncMsg(
 		commitPoint = lc.channelState.RemoteNextRevocation
 	}
 
-	if commitPoint != nil &&
-		!commitPoint.IsEqual(msg.LocalUnrevokedCommitPoint) {
+	// If we have a valid commitment point to check, then we'll verify it
+	// here. We do this unconditionally, as even those with the new
+	// tweakless format need a way to signal that they need the channel
+	// closed at state zero.
+	if commitPoint != nil && !commitPoint.IsEqual(msg.LocalUnrevokedCommitPoint) {
```

Comment:
Can we actually be checking this? The spec says 
```
MUST set `my_current_per_commitment_point` to a valid point
```
Not that it must be set to the latest unrevoked one.

Reply:
We did it unconditionally before. A prior version wouldn't check if it was tweakless, but this is wrong as we still need to signal to the upgraded party that we've lost state when we're at state zero. If we don't check this, then it looks to the up to date party, that the other party lost no data since there're checks earlier that we skip for state zero. 

For state zero against `lnd`, we purposefully set the wrong value so we can trigger a force close. many of the other implementations will auto force close based on the error we send out so I think we have that base covered.

Reply:
The case I'm still not sure about is where we are at state `n` and everything above checks out. If other implementations send a static commit point we cannot compare it to our `RemoteNextRevocation` since they won't match. Similarly, if other implementations send a non-static point, we cannot compare to a static point.

From the spec:
> `option_static_remotekey` removes the changing `to_remote` key,
> so the `my_current_per_commitment_point` is unnecessary and thus
> ignored (for parsing simplicity, it remains and must be a valid point,
> however), but the disclosure of previous secret still allows
> fall-behind detection.  An implementation can offer both, however, and
> fall back to the `option_data_loss_protect` behavior if
> `option_static_remotekey` is not negotiated.

Reply:
Resolved by having us advertise an incorrect next height. Also realized that the change to send the new error will enable the current/new recovery protocol to work better with CL as well.